torch/distributions/geometric.py

```diff
@@ -0,0 +1,71 @@
+from numbers import Number
+
+import torch
+from torch.distributions import constraints
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all, probs_to_logits, logits_to_probs, lazy_property, _finfo
+from torch.nn.functional import binary_cross_entropy_with_logits
+
+
+class Geometric(Distribution):
+    r"""
+    Creates a Geometric distribution parameterized by `probs`, where `probs` is the probability of success of Bernoulli
+    trials. It represents the probability that in k + 1 Bernoulli trials, the first k trials failed, before
+    seeing a success.
+
+    Samples are non-negative integers [0, inf).
+
+    Example::
+
+        >>> m = Geometric(torch.Tensor([0.3]))
+        >>> m.sample()  # underlying Bernoulli has 30% chance 1; 70% chance 0
+         2
+        [torch.FloatTensor of size 1]
+
+    Args:
+        probs (Number, Tensor or Variable): the probabilty of sampling `1`. Must be in range (0, 1]
+        logits (Number, Tensor or Variable): the log-odds of sampling `1`.
+    """
+    params = {'probs': constraints.unit_interval}
+    support = constraints.nonnegative_integer
+
+    def __init__(self, probs=None, logits=None):
+        if (probs is None) == (logits is None):
+            raise ValueError("Either `probs` or `logits` must be specified, but not both.")
+        if probs is not None:
+            self.probs, = broadcast_all(probs)
+            assert self.probs.gt(0).all(), 'All elements of probs must be greater than 0'
```

Comment:
Prefer `raise ValueError` for user-facing errors. `assert` is mainly for developer-facing errors.